float32 dx
float32 dy
float32 dtheta
float32[9] covariance
uint16 source_id